### ISSUE TITLE
external channel failed: timeout

### ISSUE THREAD

**Issue by galexrt** [cockpit-project/cockpit#6280]
## Log
This issue appears on a fresh installed Fedora 25 server edition.
The full log when running `/usr/libexec/cockpit-ws` is:
```
# /usr/libexec/cockpit-ws 
cockpit-bridge-Message: http://packages/shell/images/server-small.png: couldn't connect: Connection timed out
cockpit-bridge-Message: http://packages/base1/cockpit.min.js.map: couldn't connect: Connection timed out
cockpit-ws-Message: /shell/images/server-small.png: external channel failed: timeout
cockpit-ws-Message: /base1/cockpit.min.js.map: external channel failed: timeout
cockpit-bridge-Message: http://packages/shell/index.css.map: couldn't connect: Connection timed out
cockpit-ws-Message: /shell/index.css.map: external channel failed: timeout
cockpit-bridge-Message: http://packages/system/system.css: couldn't connect: Connection timed out
cockpit-bridge-Message: http://packages/system/system.js: couldn't connect: Connection timed out
cockpit-ws-Message: /system/system.css: external channel failed: timeout
cockpit-ws-Message: /system/system.js: external channel failed: timeout
cockpit-bridge-Message: http://packages/login/index.html: couldn't connect: Connection timed out
cockpit-ws-Message: /login/index.html: external channel failed: timeout
cockpit-bridge-Message: http://packages/base1/fonts/fontawesome.woff: couldn't connect: Connection timed out
cockpit-bridge-Message: http://packages/base1/fonts/patternfly.woff: couldn't connect: Connection timed out
cockpit-ws-Message: /base1/fonts/fontawesome.woff: external channel failed: timeout
cockpit-ws-Message: /base1/fonts/patternfly.woff: external channel failed: timeout
cockpit-bridge-Message: http://packages/domain/domain.js: couldn't connect: Connection timed out
cockpit-ws-Message: /domain/domain.js: external channel failed: timeout
cockpit-bridge-Message: http://packages/base1/cockpit.min.js.map: couldn't connect: Connection timed out
cockpit-ws-Message: /base1/cockpit.min.js.map: external channel failed: timeout
cockpit-bridge-Message: http://packages/base1/fonts/fontawesome.woff: couldn't connect: Connection timed out
cockpit-bridge-Message: http://packages/performance/performance.min.js.map: couldn't connect: Connection timed out
cockpit-ws-Message: /base1/fonts/fontawesome.woff: external channel failed: timeout
cockpit-ws-Message: /performance/performance.min.js.map: external channel failed: timeout
cockpit-bridge-Message: http://packages/base1/fonts/patternfly.woff: couldn't connect: Connection timed out
cockpit-ws-Message: /base1/fonts/patternfly.woff: external channel failed: timeout
```
The errors appear after a successful login.
The dashboard isn't working as it fails to load the files seen in the log above.

## cockpit-ws version
```
# /usr/libexec/cockpit-ws --version
Version: 137
Protocol: 1
Authorization: crypt1
```
## cockpit-bridge version
```
# cockpit-bridge --version
Version: 137
Protocol: 1
Payloads: dbus-json3 http-stream1 http-stream2 stream fsread1 
    fsreplace1 fswatch1 fslist1 null echo websocket-stream1 
Authorization: crypt1
```

***

## Server info
**Selinux is disabled on the server.**
```
# getenforce 
Disabled
```

**Comment by petervo** [comment]
@galexrt, I wasn't able to reproduce, can you give the output of

```
cockpit-bridge --packages
```

and

```
rpm -qa | grep cockpit
```

**Comment by galexrt** [comment]
`cockpit-bridge --packages`:
```
# cockpit-bridge --packages
base1: /usr/share/cockpit/base1
dashboard: /usr/share/cockpit/dashboard
docker: /usr/share/cockpit/docker
domain: /usr/share/cockpit/realmd
network: /usr/share/cockpit/networkmanager
performance: /usr/share/cockpit/tuned
shell: /usr/share/cockpit/shell
sosreport: /usr/share/cockpit/sosreport
ssh: /usr/share/cockpit/ssh
storage: /usr/share/cockpit/storaged
system: /usr/share/cockpit/systemd
users: /usr/share/cockpit/users
checksum = b49146d640f1497df10c9f0a4a07e7d93e6cf8a6
```
`rpm -qa | grep cockpit`:
I tried withe current version from the Fedora 25 repos:
```
# rpm -qa | grep cockpit
cockpit-ws-135-1.fc25.x86_64
cockpit-system-135-1.fc25.noarch
cockpit-docker-135-1.fc25.x86_64
cockpit-dashboard-135-1.fc25.x86_64
cockpit-storaged-135-1.fc25.noarch
cockpit-135-1.fc25.x86_64
cockpit-sosreport-135-1.fc25.noarch
cockpit-networkmanager-135-1.fc25.noarch
cockpit-bridge-135-1.fc25.x86_64
```
and then installed the packages from the Fedora cockpit group repo to test with them:
```
# rpm -qa | grep cockpit
cockpit-ws-137-1.fc25.x86_64
cockpit-storaged-137-1.fc25.noarch
cockpit-system-137-1.fc25.noarch
cockpit-137-1.fc25.x86_64
cockpit-networkmanager-137-1.fc25.noarch
cockpit-dashboard-137-1.fc25.x86_64
cockpit-sosreport-137-1.fc25.noarch
cockpit-bridge-137-1.fc25.x86_64
cockpit-docker-137-1.fc25.x86_64
```

**Comment by petervo** [comment]
Very odd, every looks very normal. Can you trying installing the debug package and turning on debug logging. https://github.com/cockpit-project/cockpit/blob/master/HACKING.md#running-cockpit-processes-under-a-debugger. The results will be pretty long, you can post a http://fpaste.org/ link back here if that makes it easier.

**Comment by galexrt** [comment]
I have build cockpit from the `137` tag.
I ran cockpit with debug from source. The gdb output: https://paste.fedoraproject.org/paste/u27p6pob8VmjLx9jELHNll5M1UNdIGYhyRLivL9gydE=
The interesting thing is that I can't even login anymore now (in line 38 it says that it can't get authentication or something). The response headers don't contain a cookie (I used chrome with the no cache option).

**Comment by petervo** [comment]
Did you run autogen.sh with --prefix=/usr and make install?

**Comment by galexrt** [comment]
Yes. But I hadn't removed the packaged cockpit version before running `make install`. Should I remove the package and re-run `make install`?

**Comment by galexrt** [comment]
@petervo I have now found the cause for the issue.
The issue is **not** coming from Cockpit.
The issue was that all servers had the proc flag `/proc/sys/net/ipv4/tcp_tw_recycle` enabled. This caused the timeouts (seems to be kernel bug; see source link).
After looking more into this issue, it was found that cockpit wasn't the only application having connection timeouts on the this one server (even though all servers had the flag enabled, later on the other server also began having connection timeouts).
This is/was fixed by disabling the `/proc/sys/net/ipv4/tcp_tw_recycle` flag.

(Source: https://bbs.archlinux.org/viewtopic.php?id=224028)
